streams/src/main/java/org/apache/kafka/streams/processor/internals/namedtopology/KafkaStreamsNamedTopologyWrapper.java

```diff
@@ -319,6 +315,14 @@ private void verifyTopologyStateStore(final String topologyName, final String st
         return streamsMetadataState.getAllMetadataForStore(storeName, topologyName);
     }
 
+    /**
+     * See {@link KafkaStreams#metadataForAllStreamsClients()}
+     */
+    public Collection<StreamsMetadata> allStreamsClientsMetadataForTopology(final String topologyName) {
```

Comment:
Not needed for this PR, but I see there's an inconsistency in the function names: these two e.g. adds a `ForTopology` suffix to differentiate with the original ones in `KafkaStreams`, but others just overload the same function. When we expose them in the KIP we need to make it consistent.

Reply:
Fair, I was thinking of them as semantically different since for example in this case we are changing the meaning of the original method to return only a subset of metadata. Whereas for something like `#store`, I felt it made sense to keep the same name since the meaning of this API doesn't change when using named topologies -- you just happen to need one additional parameter in order to uniquely identify a state store since the name alone could point to multiple stores from different topologies. But it's definitely something to discuss when we get to the KIP proposal